packages/components/src/border-box-control/linked-button/hook.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from '../styles';
+import { useContextSystem, WordPressComponentProps } from '../../ui/context';
+import { useCx } from '../../utils/hooks/use-cx';
+
+import type { LinkedButtonProps } from '../types';
+
+export function useLinkedButton(
+	props: WordPressComponentProps< LinkedButtonProps, 'div' >
+) {
+	const { className, ...otherProps } = useContextSystem(
+		props,
+		'LinkedButton'
```

Comment:
This may be a naive question because I'm not yet familiar with this context system, but wouldn't unprefixed identifiers like this cause namespace collisions?

Relates to this point 👇

> Agreed, given their private nature, it shouldn't matter much one way or another. It could get a little unwieldy prefixing them e.g. `BorderControlStyleControl`. If it is an issue and better names present themselves, it will be an easy update while polishing the final PRs.

Reply:
I'm not well versed in this context system either at the moment.

You make a great point though so to avoid any issues [I've renamed all the various subcomponents](https://github.com/WordPress/gutenberg/pull/37769/commits/f321ec20850461b47965892d6b404300d973d2ad), prefixing them with the component family as suggested. This makes a few quite verbose and awkward but for now they are consistent and predictable which is also a win.